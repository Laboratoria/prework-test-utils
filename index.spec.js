const esprima = require('esprima');

const { esprima: esprimaUtils } = require('./index.js');

const ast = esprima.parse(`
console.log(6 + 8)
`);

const consoleLogArgs = esprimaUtils.getAllConsoleLogLastArgs(ast);

describe('esprima utils', () => {
  it('getAll', () => {
    let result = esprimaUtils.getAll(ast);
    expect(result).toMatchSnapshot();

    result = esprimaUtils.getAll(ast, 'BinaryExpression');
    expect(result).toMatchSnapshot();

    result = esprimaUtils.getAll(ast, (node) => node.type === 'BinaryExpression');
    expect(result).toMatchSnapshot();
  });
  it('getAllConsoleLogLastArgs', () => {
    expect(consoleLogArgs).toMatchSnapshot();
  });
  it('getNestedExpressions,', () => {
    let result = esprimaUtils.getNestedExpressions(consoleLogArgs, 'VariableDeclaration');
    expect(result).toMatchSnapshot();

    result = esprimaUtils.getNestedExpressions(consoleLogArgs, 'ConditionalExpression');
    expect(result).toMatchSnapshot();
  });
  it('getNestedBinaryExpressions,', () => {
    const result = esprimaUtils.getNestedBinaryExpressions(consoleLogArgs, 'VariableDeclaration');
    expect(result).toMatchSnapshot();
  });
  it('getNestedUnaryExpressions,', () => {
    const result = esprimaUtils.getNestedUnaryExpressions(consoleLogArgs, 'VariableDeclaration');
    expect(result).toMatchSnapshot();
  });
});
