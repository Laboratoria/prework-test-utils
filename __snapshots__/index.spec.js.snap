// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`esprima utils getAll 1`] = `
Array [
  Program {
    "body": Array [
      ExpressionStatement {
        "expression": CallExpression {
          "arguments": Array [
            BinaryExpression {
              "left": Literal {
                "raw": "6",
                "type": "Literal",
                "value": 6,
              },
              "operator": "+",
              "right": Literal {
                "raw": "8",
                "type": "Literal",
                "value": 8,
              },
              "type": "BinaryExpression",
            },
          ],
          "callee": StaticMemberExpression {
            "computed": false,
            "object": Identifier {
              "name": "console",
              "type": "Identifier",
            },
            "property": Identifier {
              "name": "log",
              "type": "Identifier",
            },
            "type": "MemberExpression",
          },
          "type": "CallExpression",
        },
        "type": "ExpressionStatement",
      },
    ],
    "sourceType": "script",
    "type": "Program",
  },
  ExpressionStatement {
    "expression": CallExpression {
      "arguments": Array [
        BinaryExpression {
          "left": Literal {
            "raw": "6",
            "type": "Literal",
            "value": 6,
          },
          "operator": "+",
          "right": Literal {
            "raw": "8",
            "type": "Literal",
            "value": 8,
          },
          "type": "BinaryExpression",
        },
      ],
      "callee": StaticMemberExpression {
        "computed": false,
        "object": Identifier {
          "name": "console",
          "type": "Identifier",
        },
        "property": Identifier {
          "name": "log",
          "type": "Identifier",
        },
        "type": "MemberExpression",
      },
      "type": "CallExpression",
    },
    "type": "ExpressionStatement",
  },
  CallExpression {
    "arguments": Array [
      BinaryExpression {
        "left": Literal {
          "raw": "6",
          "type": "Literal",
          "value": 6,
        },
        "operator": "+",
        "right": Literal {
          "raw": "8",
          "type": "Literal",
          "value": 8,
        },
        "type": "BinaryExpression",
      },
    ],
    "callee": StaticMemberExpression {
      "computed": false,
      "object": Identifier {
        "name": "console",
        "type": "Identifier",
      },
      "property": Identifier {
        "name": "log",
        "type": "Identifier",
      },
      "type": "MemberExpression",
    },
    "type": "CallExpression",
  },
  StaticMemberExpression {
    "computed": false,
    "object": Identifier {
      "name": "console",
      "type": "Identifier",
    },
    "property": Identifier {
      "name": "log",
      "type": "Identifier",
    },
    "type": "MemberExpression",
  },
  Identifier {
    "name": "console",
    "type": "Identifier",
  },
  Identifier {
    "name": "log",
    "type": "Identifier",
  },
  BinaryExpression {
    "left": Literal {
      "raw": "6",
      "type": "Literal",
      "value": 6,
    },
    "operator": "+",
    "right": Literal {
      "raw": "8",
      "type": "Literal",
      "value": 8,
    },
    "type": "BinaryExpression",
  },
  Literal {
    "raw": "6",
    "type": "Literal",
    "value": 6,
  },
  Literal {
    "raw": "8",
    "type": "Literal",
    "value": 8,
  },
]
`;

exports[`esprima utils getAll 2`] = `
Array [
  BinaryExpression {
    "left": Literal {
      "raw": "6",
      "type": "Literal",
      "value": 6,
    },
    "operator": "+",
    "right": Literal {
      "raw": "8",
      "type": "Literal",
      "value": 8,
    },
    "type": "BinaryExpression",
  },
]
`;

exports[`esprima utils getAll 3`] = `
Array [
  BinaryExpression {
    "left": Literal {
      "raw": "6",
      "type": "Literal",
      "value": 6,
    },
    "operator": "+",
    "right": Literal {
      "raw": "8",
      "type": "Literal",
      "value": 8,
    },
    "type": "BinaryExpression",
  },
]
`;

exports[`esprima utils getAllConsoleLogLastArgs 1`] = `
Array [
  BinaryExpression {
    "left": Literal {
      "raw": "6",
      "type": "Literal",
      "value": 6,
    },
    "operator": "+",
    "right": Literal {
      "raw": "8",
      "type": "Literal",
      "value": 8,
    },
    "type": "BinaryExpression",
  },
]
`;

exports[`esprima utils getNestedBinaryExpressions, 1`] = `
Array [
  BinaryExpression {
    "left": Literal {
      "raw": "6",
      "type": "Literal",
      "value": 6,
    },
    "operator": "+",
    "right": Literal {
      "raw": "8",
      "type": "Literal",
      "value": 8,
    },
    "type": "BinaryExpression",
  },
]
`;

exports[`esprima utils getNestedExpressions, 1`] = `Array []`;

exports[`esprima utils getNestedExpressions, 2`] = `Array []`;

exports[`esprima utils getNestedUnaryExpressions, 1`] = `Array []`;
